static/gsApp/hooks/useBillingDetails.tsx

```diff
@@ -0,0 +1,42 @@
+import {
+  keepPreviousData,
+  useApiQuery,
+  type QueryObserverResult,
+} from 'sentry/utils/queryClient';
+import type RequestError from 'sentry/utils/requestError/requestError';
+import useOrganization from 'sentry/utils/useOrganization';
+
+import type {BillingDetails} from 'getsentry/types';
+
+interface HookResult {
+  data: BillingDetails | undefined;
+  error: RequestError | null;
+  isError: boolean;
+  isLoading: boolean;
+  refetch: () => Promise<QueryObserverResult<BillingDetails, unknown>>;
+}
+
+export function useBillingDetails(): HookResult {
+  const organization = useOrganization();
+
+  const {data, isPending, isLoading, isError, error, refetch} =
+    useApiQuery<BillingDetails>([`/customers/${organization.slug}/billing-details/`], {
```

Comment:
might be easier to return the entire thing and you could remove the "HookResult" type
```suggestion
    return useApiQuery<BillingDetails>([`/customers/${organization.slug}/billing-details/`], {
```